src/Admin/UserRESTEndpointExtension.php

```diff
@@ -0,0 +1,106 @@
+/**
+ * Class UserRESTEndpointExtension
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * UserRESTEndpointExtension class.
+ *
+ * @since 2.2
+ * @internal
+ */
+class UserRESTEndpointExtension implements Service, Registerable, Delayed {
+
+	/**
+	 * User meta key that stores a template mode for which the "Review" panel was dismissed.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE = 'amp_review_panel_dismissed_for_template_mode';
+
+	/**
+	 * Get registration action.
+	 *
+	 * @return string
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Adds hooks.
+	 */
+	public function register() {
+		$this->register_rest_field();
+	}
+
+	/**
+	 * Register REST field for storing a template mode for which the "Review" panel was dismissed.
+	 */
+	public function register_rest_field() {
+		register_rest_field(
+			'user',
+			self::USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE,
+			[
+				'get_callback'    => [ $this, 'get_review_panel_dismissed_for_template_mode' ],
+				'update_callback' => [ $this, 'update_review_panel_dismissed_for_template_mode' ],
+				'schema'          => [
+					'description' => __( 'The template mode for which the Review panel on the Settings screen was dismissed by a user.', 'amp' ),
+					'type'        => 'string',
+					'enum'        => [
+						'',
+						AMP_Theme_Support::READER_MODE_SLUG,
+						AMP_Theme_Support::STANDARD_MODE_SLUG,
+						AMP_Theme_Support::TRANSITIONAL_MODE_SLUG,
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Provides a template mode for which the "Review" panel has been dismissed by a user.
+	 *
+	 * @param array $user Array of user data prepared for REST.
+	 *
+	 * @return string|null Template mode for which the panel is dismissed, empty string if the option has not been set.
+	 */
+	public function get_review_panel_dismissed_for_template_mode( $user ) {
+		return get_user_meta( $user['id'], self::USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE, true );
+	}
+
+	/**
+	 * Updates a user's setting determining for which template mode the "Review" panel was dismissed.
+	 *
+	 * @param string  $template_mode Template mode or empty string.
+	 * @param WP_User $user          The WP user to update.
+	 *
+	 * @return bool|WP_Error The result of update_user_meta, or WP_Error if the current user lacks permission.
+	 */
+	public function update_review_panel_dismissed_for_template_mode( $template_mode, WP_User $user ) {
+		if ( ! current_user_can( 'edit_user', $user->ID ) ) {
+			return new WP_Error(
+				'amp_rest_cannot_edit_user',
+				__( 'Sorry, the current user is not allowed to make this change.', 'amp' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
+		}
+
```

Comment:
I also realized that this is not needed either, because the `update_callback` will not run for the user endpoint if the user cannot `edit_user` thus the Users controller will return false when `update_item_permissions_check` is called.